tools/keytools/keygen.c

```diff
@@ -356,6 +358,66 @@ static int generated_keypairs_type[MAX_KEYPAIRS];
 static uint32_t generated_keypairs_id_mask[MAX_KEYPAIRS];
 static int n_generated = 0;
 
+static char *append_pub_to_fname(const char *filename) {
+    const char pubSuffix[] = "_pub";
+    const size_t pubSuffixLength = strlen(pubSuffix);
+
+    const char *dotPosition = strchr(filename, '.');
+
+    size_t prefixLength;
+    if (dotPosition != NULL) {
+        prefixLength = dotPosition - filename;
+    } else {
+        prefixLength = strlen(filename);
+    }
+
+    size_t newFilenameLength = prefixLength + pubSuffixLength + (dotPosition ? strlen(dotPosition) : 0) + 1;
```

Comment:
Please avoid mixed-dec. These tools need to be super portable.

Reply:
done